### PR TITLE
convert cli/domain to pytest

### PR DESCRIPTION
**Pull request by peterdragun**
Test results:
```
pytest -v tests/foreman/cli/test_domain.py 
=============================================== test session starts ===============================================
platform linux -- Python 3.7.8, pytest-4.6.3, py-1.8.0, pluggy-0.12.0 -- /home/pdragun/.virtualenvs/robottelo_master/bin/python3
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pdragun/Documents/robottelo
plugins: forked-1.0.2, xdist-1.29.0, mock-1.10.4, cov-2.7.1, services-1.3.1
collecting ... 2020-08-31 11:51:47 - conftest - DEBUG - Collected 12 test cases
collected 12 items                                                                                                

tests/foreman/cli/test_domain.py::TestDomain::test_positive_create_update_delete_domain PASSED              [  8%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_create[0] PASSED                                [ 16%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_create_with_invalid_dns_id PASSED               [ 25%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_update[0] PASSED                                [ 33%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_update[1] PASSED                                [ 41%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_set_parameter[0] PASSED                         [ 50%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_set_parameter[1] PASSED                         [ 58%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_set_parameter[2] PASSED                         [ 66%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_delete_by_id[0] PASSED                          [ 75%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_delete_by_id[1] PASSED                          [ 83%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_delete_by_id[2] PASSED                          [ 91%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_delete_by_id[3] PASSED                          [100%]

===================================== 12 passed, 5 warnings in 147.70 seconds =====================================
```

### CODE REVIEW

File: tests/foreman/cli/test_domain.py [hunk 1]
```diff
@@ -113,9 +114,14 @@ def valid_delete_params():
...
 
 
-class DomainTestCase(CLITestCase):
+class TestDomain:
     """Domain CLI tests"""
 
+    @pytest.fixture(scope="class")
+    def class_domain(self):
+        """Create shared domain"""
+        return make_domain()
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_domain.py:123]: How would you like to try stripping these out of the class entirely and making this a module-level fixture?

**Review by rplevka — CHANGES_REQUESTED** [review]
Could you setup a clean, up-to-date robottelo environment and re-run those tests?
I recently had to revert a commit, which used parameterized tests inside a test class.
It looked like one was not able to use parametrize decorator on a test level in tests nested in classes, but your commit seems to indicate the opposite.

Thank you!

**Comment by peterdragun** [comment]
Updated to use module shared fixture for domain.
@rplevka I have created new clean venv and got similar test results as before: 
```
 pytest -v tests/foreman/cli/test_domain.py 
========================================= test session starts ==========================================
platform linux -- Python 3.7.8, pytest-4.6.3, py-1.9.0, pluggy-0.13.1 -- /home/pdragun/.virtualenvs/fresh_robottelo/bin/python3
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pdragun/Documents/robottelo
plugins: mock-1.10.4, services-1.3.1
collecting ... 2020-09-02 13:13:07 - conftest - DEBUG - Collected 12 test cases
collected 12 items                                                                                     

tests/foreman/cli/test_domain.py::TestDomain::test_positive_create_update_delete_domain PASSED   [  8%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_create[0] PASSED                     [ 16%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_create_with_invalid_dns_id PASSED    [ 25%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_update[0] PASSED                     [ 33%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_update[1] PASSED                     [ 41%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_set_parameter[0] PASSED              [ 50%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_set_parameter[1] PASSED              [ 58%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_set_parameter[2] PASSED              [ 66%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_delete_by_id[0] PASSED               [ 75%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_delete_by_id[1] PASSED               [ 83%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_delete_by_id[2] PASSED               [ 91%]
tests/foreman/cli/test_domain.py::TestDomain::test_negative_delete_by_id[3] PASSED               [100%]

=============================== 12 passed, 5 warnings in 175.86 seconds ================================
```

**Comment by JacobCallahan** [comment]
@rplevka brings up a good point. Now that we're moving away from unittest, we don't necessarily need to wrap these tests in a class. @peterdragun can you try removing the class and bringing these tests to the module level?

**Comment by peterdragun** [comment]
Class removed.
New test results:
```
pytest -v tests/foreman/cli/test_domain.py 
========================================= test session starts ==========================================
platform linux -- Python 3.7.8, pytest-4.6.3, py-1.9.0, pluggy-0.13.1 -- /home/pdragun/.virtualenvs/fresh_robottelo/bin/python3
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pdragun/Documents/robottelo
plugins: mock-1.10.4, services-1.3.1
collecting ... 2020-09-03 08:55:21 - conftest - DEBUG - Collected 12 test cases
collected 12 items                                                                                     

tests/foreman/cli/test_domain.py::test_positive_create_update_delete_domain PASSED               [  8%]
tests/foreman/cli/test_domain.py::test_negative_create[0] PASSED                                 [ 16%]
tests/foreman/cli/test_domain.py::test_negative_create_with_invalid_dns_id PASSED                [ 25%]
tests/foreman/cli/test_domain.py::test_negative_update[0] PASSED                                 [ 33%]
tests/foreman/cli/test_domain.py::test_negative_update[1] PASSED                                 [ 41%]
tests/foreman/cli/test_domain.py::test_negative_set_parameter[0] PASSED                          [ 50%]
tests/foreman/cli/test_domain.py::test_negative_set_parameter[1] PASSED                          [ 58%]
tests/foreman/cli/test_domain.py::test_negative_set_parameter[2] PASSED                          [ 66%]
tests/foreman/cli/test_domain.py::test_negative_delete_by_id[0] PASSED                           [ 75%]
tests/foreman/cli/test_domain.py::test_negative_delete_by_id[1] PASSED                           [ 83%]
tests/foreman/cli/test_domain.py::test_negative_delete_by_id[2] PASSED                           [ 91%]
tests/foreman/cli/test_domain.py::test_negative_delete_by_id[3] PASSED                           [100%]

=============================== 12 passed, 5 warnings in 163.50 seconds ================================
```

**Review by rplevka — APPROVED** [review]

**Review by JacobCallahan — APPROVED** [review]
ACK

**Pull request merged**